### PR TITLE
Provenance_Histos_Renormalization_81X_V1

### PR DESCRIPTION
**Pull request by archiron**
A new normalization was made for "provenance (*)" histos for standard histos (SingleElectronPt10, 35, .., TTbar, ZEE) and QCD_80_120 in ElectronMcSignalPostValidator.cc and ElectronMcFakePostValidator.cc files.
No other modification was made.

(*) h1_ele_provenance, h1_ele_provenance_barrel, h1_ele_provenance_endcaps
@beaudett @fcouderc

### CODE REVIEW

File: Validation/RecoEgamma/plugins/ElectronMcFakePostValidator.cc [hunk 1]
```diff
@@ -59,6 +59,14 @@ void ElectronMcFakePostValidator::finalize( DQMStore::IBooker & iBooker, DQMStor
     h1_ele_xOverX0VsEta->setBinContent(ibin,xOverX0) ;
   }
 /**/
+
+  MonitorElement * h1_ele_provenance = get(iGetter, "provenance") ;
+  h1_ele_provenance->getTH1F()->Scale(1./h1_ele_provenance->getBinContent(3));
+  MonitorElement * h1_ele_provenance_barrel = get(iGetter, "provenance_barrel") ;
+  h1_ele_provenance_barrel->getTH1F()->Scale(1./h1_ele_provenance_barrel->getBinContent(3));
+  MonitorElement * h1_ele_provenance_endcaps = get(iGetter, "provenance_endcaps") ;
+  h1_ele_provenance_endcaps->getTH1F()->Scale(1./h1_ele_provenance_endcaps->getBinContent(3));*/
```
Comment (dmitrijus) [inline Validation/RecoEgamma/plugins/ElectronMcFakePostValidator.cc]: The trailing */ looks like an error.

**Review by dmitrijus — CHANGES_REQUESTED** [review]

**Comment by dmitrijus** [comment]
80x: https://github.com/cms-sw/cmssw/pull/16247

Please reference pull requests like I just did. Saves a lot of time for a number of people.

**Comment by dmitrijus** [comment]
+1

**Comment by davidlange6** [comment]
+1

File: Validation/RecoEgamma/plugins/ElectronMcFakePostValidator.cc [hunk 2]
```diff
@@ -59,6 +59,14 @@ void ElectronMcFakePostValidator::finalize( DQMStore::IBooker & iBooker, DQMStor
     h1_ele_xOverX0VsEta->setBinContent(ibin,xOverX0) ;
   }
 /**/
+
+  MonitorElement * h1_ele_provenance = get(iGetter, "provenance") ;
+  h1_ele_provenance->getTH1F()->Scale(1./h1_ele_provenance->getBinContent(3));
+  MonitorElement * h1_ele_provenance_barrel = get(iGetter, "provenance_barrel") ;
+  h1_ele_provenance_barrel->getTH1F()->Scale(1./h1_ele_provenance_barrel->getBinContent(3));
+  MonitorElement * h1_ele_provenance_endcaps = get(iGetter, "provenance_endcaps") ;
+  h1_ele_provenance_endcaps->getTH1F()->Scale(1./h1_ele_provenance_endcaps->getBinContent(3));
```
Comment (slava77) [inline Validation/RecoEgamma/plugins/ElectronMcFakePostValidator.cc]: Looks like these changes has broken plots/comparisons done with jenkins.
Low stat tests have the bin3 content 0 and these histograms become nans.
Given that "nan!=nan" in floating point logic, the histograms with nans are reported as different.

Please review the changes and avoid divisions by 0

Comment (beaudett) [inline Validation/RecoEgamma/plugins/ElectronMcFakePostValidator.cc]: Hi Slava,

as a matter of fact, we usually run our tests on as much stat as possible, we were not aware of this use case, sorry.

Can you please confirm that in the low stat test that you mention the histograms are simply empty ? In principle, the bin3 should always be filled as soon as there is an electron candidate. 
If it is indeed the case, we will simply skip the rescaling when the 3rd bin content is empty and it won't matter because the histogram will be empty anyway. I would not feel comfortable to apply different normalization in different cases. 
Thanks

Comment (slava77) [inline Validation/RecoEgamma/plugins/ElectronMcFakePostValidator.cc]: I checked ``` /DQMData/Run 1/EgammaV/Run summary/ElectronMcFakeValidator ``` h_ele_provenance
the statbox says entries = 0.

Comment (beaudett) [inline Validation/RecoEgamma/plugins/ElectronMcFakePostValidator.cc]: Thanks Slava. We'll fix that early next week.

Comment (archiron) [inline Validation/RecoEgamma/plugins/ElectronMcFakePostValidator.cc]: Hi, I made corrections on the files on Provenance_Histos_Renormalization_81X_V1 #16246
Do I have to made another PR ?

Comment (slava77) [inline Validation/RecoEgamma/plugins/ElectronMcFakePostValidator.cc]: yes, please.
